.moban.yaml

```diff
@@ -65,3 +67,6 @@ targets:
   - netlify.toml: docs/netlify.toml
   - .misc/appveyor.yml: ci/appveyor.yml.jj2
   - .misc/run_with_env.cmd: run_with_env.cmd
+copy:
+  # homeassistant license MIT
+  - coalib/misc/Asyncio.py: async.py
```

Comment:
What is the benefit of using `moban` to copy `async.py` from https://github.com/NAStools/homeassistant/blob/master/homeassistant/util/async.py? That project seems to be out of maintenance.

Drawbacks I can see:
1. licence is not stated in `async.py`, but in this file.
2. `async.py` needs to be excluded in `.coafile`.
3. `async.py` has other functions, e.g. `fire_coroutine_threadsafe`, which we don't use.
4. We cannot easily test it.

Reply:
The reason to do it this way is so that external code is 'managed'.  We can improve the way it is managed by improving moban.

Your point '3' is the only significant problem, and that can be solved by moban having patch support, so we can exclude the other code e.g. https://github.com/moremoban/moban/issues/75 . A header can also be added using a patch.

I am not worried about the licensing being missing from the file, as anyone modifying this file will likely be informed via code review, or will be reverted when someone looks at the moban job.  We should distribute the MIT license, and any other license of code we're including, but by no means is this "substantial portions" according to the MIT license so we're not breaking the license by not including it.  And by having a control file which mentions where code came from, and thus were the license text can be found, we're not hiding the fact the code is copied.

The fact it is unmaintained is a problem for the future, as PRs upstream will not get merged.  Again, a .patch could solve that, but i would expect that this type of code would be moved to a new `coala-compat` package, or something like that, or a sub-module of coala-utils.

Same with the .coafile problems.  A patch can fix it.

The drawbacks of not doing it:

- another piece of copied code will fade into the coala codebase and its history effectively lost
- coala doesnt have a process for allowing externally written code into the repo.
- moban doesnt have a user (coala) pushing for it to handle more complicated scenarios than just 'copy'.
- coala doesnt start the process of identifying and fixing any existing copied code.  This one is MIT, but we will likely find some which isnt so freely licensed.  e.g. we have CC-BY-SA-licensed 'code' in some repos, and a major issue over in coala-quickstart
- etc.

Being in `.moban.yaml` is an annoyance.  That is intentional ;-)